xenmgr/Vm/Actions.hs

```diff
@@ -775,20 +775,6 @@ setupAcpiNode uuid =
      case stubdom of
          Just stubdomid -> liftIO $ xsChmod (xsp domid ++ "/acpi-state") ("b" ++ show stubdomid)
          Nothing        -> liftIO $ xsChmod (xsp domid ++ "/acpi-state") ("b" ++ show domid)
- 
---Watch acpi state when booting a VM, used to be handled in xenvm
```

Comment:
Removing this would also drop the ```powerlinkR``` stuff, which is supposed to change the host state based on guest state.  So if you sleep the windows VM, it was supposed to sleep the host.  Whether or not that's working right now is unknown, and whether we want to continue to 'support' it is also up for debate.

You're right for the most part about the rest of the state change stuff, the important callbacks are fired based off vm state, not acpiState, like ```set_shr``` which is needed at the moment to differentiate between reboot and shutdown.  However one thing it is useful for is performing an action when the guest sleeps or resumes from sleep, since the VM state is always considered 'running'. Only the acpiState denotes that it is asleep/awake.

Reply:
We do have guest S3 working, we'll try and make sure whatever is making that work gets upstreamed soon.  

This PR got away from me, I'll need to look through it again but I don't disagree with what you've said here.  I'm definitely not married to the powerlinkR stuff, although it would be nice to evaluate whether we have any more luck with host S3 on a newer xen version.  I've completely forgotten the reason why it doesn't work today.

Reply:
You wrote:

> We do have guest S3 working, we'll try and make sure whatever is making that work gets upstreamed soon.

When you say guest, you mean VM?  I note these two failures in the PR description:

> Also, VM S3 seems broken. Windows 10 can't S3 as an OpenXT VM because the graphics subsystem prevents it. However, I haven't checked if the vglass drivers allow it. Linux detaches its PV disks on S3 resume, but the backend doesn't reconnect so it is basically dead.

Windows S3 disabling (shown by `powercfg /a`) is with emulated VGA.  Maybe it works for you with the vGlass PV drivers?